projects/ad4110/src/app/main.c

```diff
@@ -0,0 +1,140 @@
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "irq.h"
+#include "irq_extra.h"
+#include "gpio_irq_extra.h"
+#include "spi.h"
+#include "spi_extra.h"
+#include "ad4110.h"
+#include "error.h"
+#include "xparameters.h"
+#include "parameters.h"
+#include <stdint.h>
+
+int32_t data_buf[BUF_LENGTH];
+
+/***************************************************************************//**
+ * @brief main
+*******************************************************************************/
+int main()
+{
+	uint32_t ret;
+
+	struct xil_spi_init_param spi_extra = {
+		.type = SPI_PS,
+		.flags = 0U
+	};
+	struct spi_init_param spi_ip = {
+		.device_id = SPI_DEVICE_ID,
+		.max_speed_hz = 100000,
+		.mode = SPI_MODE_0,
+		.chip_select = 0U,
+		.bit_order = SPI_BIT_ORDER_MSB_FIRST,
+		.platform_ops = &xil_spi_ops,
+		.extra = &spi_extra
+	};
+
+	/* IRQ instance. */
+	struct irq_ctrl_desc *irq_desc;
+	struct xil_irq_init_param irq_extra = {
+		.type = IRQ_PS
+	};
+	struct irq_init_param irq_ip = {
+		.irq_ctrl_id = 0U,
+		.platform_ops = &xil_irq_ops,
+		.extra = &irq_extra
+	};
+
+	ret = irq_ctrl_init(&irq_desc, &irq_ip);
+	if(ret)
+		return FAILURE;
+	ret = irq_global_enable(irq_desc);
+	if(ret)
+		return FAILURE;
+
+	/* GPIO IRQ instance. */
+	struct irq_ctrl_desc *gpio_irq_desc;
+	struct xil_gpio_irq_init_param gpio_irq_extra = {
+		.parent_desc = irq_desc,
+		.gpio_device_id = XPAR_PS7_GPIO_0_DEVICE_ID,
+		.retriggerable = false
```

Comment:
Hasn't this be removed?

Reply:
This is the gpio irq descriptor, it is needed by adc_continuous_read function. Do not confuse this with gpio pin that was removed from the driver.

Reply:
I talk about the retriggerable parameter.
The decided aproach retrigger by default and to be able to disable it using irq_disable